### PR TITLE
Test Cases for helper functions in datepicker.js #1939

### LINKED ISSUE #1939: Add test Cases for helper functions in datepicker.js
4 new functions have been added to the datepicker.js file and need JS test cases.

All are very simple to be tested.

### PR DESCRIPTION
**Pull request by nsaiisasidhar**
Fixes #1939 

Please review and approve the PR, if the test cases are good and complete.

### CODE REVIEW

**Comment by junhaoyap** [comment]
Generally ok, @thyageshm can you provide some input as I am not familiar with the functions

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java [hunk 1]
```diff
@@ -355,7 +355,7 @@ public void testAddAction() throws Exception {
         savedSession = BackDoor.getFeedbackSession(newSession.getCourseId(), newSession.getFeedbackSessionName());
         newSession.sanitizeForSaving();
 
-        newSession.setInstructions(new Text("<p>cannot see responses</p>\r\n\r\n<p>$^/&#61;?</p>"));
```
Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java]: Any reason for this edit? I am not sure if this is related to this fix

File: src/test/javascript/DatepickerJsTest.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+QUnit.test('getMaxDateForStartDate(endDate)', function(assert) {
+    assert.deepEqual(getMaxDateForStartDate(newDate('02 Apr 2012, 23:59')), newDate('02 Apr 2012, 23:59'),
+                     'Return endDate input argument');
+});
+
+QUnit.test('getMaxDateForVisibleDate(startDate, publishDate)', function(assert) {
+    assert.deepEqual(getMaxDateForVisibleDate(newDate('02 Apr 2012, 23:59'), null), newDate('02 Apr 2012, 23:59'),
+                     'NULL value test');
+    assert.deepEqual(getMaxDateForVisibleDate(newDate('02 Apr 2012, 23:59'), ''), newDate('02 Apr 2012, 23:59'),
```
Comment (thyageshm) [inline src/test/javascript/DatepickerJsTest.js]: I dont think ' ' is undefined. I think we should literally pass in *undefined*

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java (hunk 1 above)
Comment (nsaiisasidhar) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackPageUiTest.java]: @thyageshm That is to revert changes made to InstructorFeedbackPageUiTest.java. I'll remove it.

**Comment by thyageshm** [comment]
@junhaoyap I am done

**Comment by junhaoyap** [comment]
Thank you so much @thyageshm!

**Comment by junhaoyap** [comment]
Thank you @nsaiisasidhar, great work!

**Comment by nsaiisasidhar** [comment]
That's fine. Thanks for your reviews and support.

**Comment by damithc** [comment]
This seems to be a new file containing tests. Shouldn't it be added to the test suite in some other place? @li-kai can have a look (because you are looking into JS tests these days)?

**Comment by li-kai** [comment]
@damithc The test is placed where it should be. However, the test file path should be added to `src/test/resources/pages/allJsUnitTests.html`.

**Comment by junhaoyap** [comment]
Lucky to have more eyes as I did not notice that it wasn’t added into the html file

**Comment by junhaoyap** [comment]
@nsaiisasidhar, some changes are requested :)

**Comment by junhaoyap** [comment]
@nsaiisasidhar unfortunately it seems that not that the tests are being run they don't pass, could we take some time to look at exactly which of the test that were added does not pass?

File: src/test/javascript/DatepickerJsTest.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+QUnit.module('datepicker.js');
+
+QUnit.test('getMinDateForEndDate(startDate)', function(assert) {
+    assert.deepEqual(getMinDateForEndDate(newDate('02/04/12')), newDate('02/04/12'),
```
Comment (wkurniawan07) [inline src/test/javascript/DatepickerJsTest.js]: Is `newDate` a function defined somewhere or are you trying to use `new Date`?

**Comment by wkurniawan07** [comment]
@nsaiisasidhar may I suggest that you try and get the test to pass in your computer first, instead of relying on Travis CI?

**Comment by wkurniawan07** [comment]
Closing due to inactivity.

**Pull request closed without merging**